lib/services/role.go

```diff
@@ -2332,6 +2332,13 @@ type databaseUserMatcher struct {
 // NewDatabaseUserMatcher creates a RoleMatcher that checks whether the role's
 // database users match the specified condition.
 func NewDatabaseUserMatcher(db types.Database, user string) RoleMatcher {
+	if db.SupportAWSIAMRolesAsUsers() {
+		return &databaseUserMatcher{
+			user:             user,
+			alternativeNames: makeUsernamesForAWSRole(db, user),
+		}
+	}
```

Comment:
nit: the logic looks good to me. maybe we can improve the naming to differentiate between `makeAlternativeNamesForAWSRole`, something like
```suggestion
	if db.SupportAWSIAMRoleARNsAsUsers() {
		return &databaseUserMatcher{
			user:             user,
			alternativeNames: makeUsernamesForAWSRoleARN(db, user),
		}
	}
```